core/src/main/java/feign/FeignException.java

```diff
@@ -142,13 +155,24 @@ public String contentUTF8() {
     }
   }
 
+  /**
+   * The Response headers.
+   *
+   * @return the headers from the {@link Response}.
+   * @see Response#headers()
+   */
+  public Map<String, Collection<String>> responseHeaders() {
+    return this.responseHeaders;
```

Comment:
We should also wrap this as an immutable collection.  Can we wrap with `Collections.unmodifiableMap`?

Reply:
Sure, but if we wrap the response headers as an immutable collection inside the constructor, shouldn't that be sufficient?

Reply:
I should be, but we don't advertise that to the consumers.  It's also good practice to return a copy of the internal collection just in case.

Reply:
I'd argue that it's better to do the copy on construction (and optionally wrap it in `Collections.unmodifiableMap()`) instead of returning a new copy each time.